scripts/linters/pylint_extensions.py

```diff
@@ -1854,6 +1820,133 @@ def process_tokens(self, tokens):
                         'single-line-pragma', line=line_num)
 
 
+class SingleSpaceAfterKeyWordChecker(checkers.BaseChecker):
+    """Custom pylint checker which checks that there is a single space
+    between a keyword and a bracket `(` if a bracket is used. Nodes
+    tested are `If`, `IfExp`, `While`, and `Yield`.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+
+    name = 'single-space-after-if'
+    priority = -1
+    msgs = {
+        'C0029': (
+            'Not using a single space after `if` statement.',
+            'single-space-after-if',
+            'Add a single space after the `if` statement.',
+        ),
+    }
+    name = 'single-space-after-elif'
+    priority = -1
+    msgs = {
+        'C0030': (
+            'Not using a single space after `elif` statement.',
+            'single-space-after-elif',
+            'Add a single space after the `elif` statement.',
+        ),
+    }
+    name = 'single-space-after-while'
+    priority = -1
+    msgs = {
+        'C0031': (
+            'Not using a single space after `while` statement.',
+            'single-space-after-while',
+            'Add a single space after the `while` statement.',
+        ),
+    }
+    name = 'single-space-after-yield'
+    priority = -1
+    msgs = {
+        'C0032': (
+            'Not using \'yield\' or a single space after `yield` statement.',
+            'single-space-after-yield',
+            'Add a single space after the `yield` statement.',
+        ),
+    }
+
+    def visit_if(self, node):
+        """Visit every `if`, and `elif` statement to make sure they are
+        followed by a single space.
+
+        Args:
+            node: astroid.nodes.If. Node to access `If` content.
+        """
+
+        # Check the `if` statement only if it has a parenthesis.
+        line_number = node.fromlineno
+        line = linecache.getline(node.root().file, line_number).strip()
+        if re.search(br'^if\s*\(', line):
+            parenthesis_index = line.find('(')
+            first_space_index = line.find(' ')
+            # There should be exactly one space between these indices.
+            if parenthesis_index - first_space_index != 1:
+                self.add_message('single-space-after-if', line=line_number)
+
+        # Check each `elif` that has a parenthesis.
+        for line_number in python_utils.RANGE(line_number, node.tolineno + 1):
```

Comment:
There should be some node for `elif`. Can you please check?

Reply:
Sure! Seems to me like [there is no `elif` node](https://astroid.readthedocs.io/en/latest/api/astroid.nodes.html). There is an [`if` node](https://astroid.readthedocs.io/en/latest/api/astroid.nodes.html#astroid.nodes.If), which only includes the function `has_elif_block()`. Is there something you see that I'm skipping over?